### ISSUE TITLE
What Kind of Arabic Should we Use?

### ISSUE THREAD

**Issue by obahareth** [reactjs/ar.reactjs.org#24]
I would like to help with the translation but was wondering what form of Arabic should we choose? Are we going to go with Modern Standard Arabic `العربية الفصحى` or something closer to the accents that people use in their daily speech (going to call it to Colloquial Arabic `العربية العامية`)? I ask this because I and others can help with translating to Colloquial Arabic but our Modern Standard Arabic is very weak and most people don't speak it even at work these days. I believe this is an important matter to discuss because what we decide here in this translation may very well become the standard upon which other translations are based.

To add some context, in most Arabic writing we use Modern Standard Arabic, and it's very uncommon to use it in a conversation at work for example; it's more popular in speeches or delivering news, books, etc. The Colloquial Arabic that most people speak nowadays is sometimes seen as a lesser form of Arabic that should be avoided.

As a very small example I see calling a function described as `استدعاء الدالة` and if I heard the word `استدعاء`, I would associate more with being summoned to court rather than calling a function, I would perhaps use something like `مناداة` if I try to take how most Arabs speak rather than write into account (I know some math/programming books do this, but do we want to use them as our base?). 

If we're going to go with Modern Standard Arabic, what are the standards? Should we use diacritics `تشكيل` everywhere or only some places for example? I think having these in the contributor guidelines would be very helpful. I think this is something good to tackle early on so the docs have the same vibe everywhere.

Final note:
I am not saying any form of Arabic is better or worse than the other, I just want to know what we're going with to understand if I can assist this effort with my limited knowledge.

**Comment by aladin002dz** [comment]
we can use, for the first version, this translation:
https://wiki.hsoub.com/React

**Comment by obahareth** [comment]
Are we just going to use it as is for the first version?

**Comment by asantarissy** [comment]
> Are we just going to use it as is for the first version?

Thank you @obahareth, yes let's have a first version for now, and then we're definitely going to proofread everything and adjust it accordingly.

**Comment by fkanout** [comment]
Could you please mention that you want to rely on https://wiki.hsoub.com/React (as a start) in the contribution guide?

**Comment by shakesBeardZ** [comment]
@obahareth it's not even an issue to raise I think we all agree as Arabic speakers that formal Arabic is the only Arabic that Arab people can understand, no offense but other accents can't be understood in other countries. take for example me as an Algerian I'll have difficulties fully understanding Egyptian or Saudi accent so the only way is to use formal one like any Arabic book.

**Comment by obahareth** [comment]
@shakesBeardZ 
> no offense but other accents can't be understood in other countries

That is very true, but the same is true with Formal Arabic as well, especially with younger audiences. I was born and raised in Saudi Arabia and I've lived most of my life there but I rarely ever get a chance to speak or write Formal Arabic, and as I mentioned before I know many who are in the same situation, and that number seems to be growing.

My intention was definitely **not** to imply that we should use a form of Colloquial Arabic specific to a certain region. I believe Modern Standard Arabic `العربية الفصحى` is the right basis for this (not necessarily formal) but I think we should try to use terms that are more well-known or more used, Which is why I gave the example of `استدعاء` vs `مناداة` (I've heard `مناداة` being used amongst Arab programmers far more than `استدعاء`, but of course I'm **not** saying we should base the translation based on what I've heard).

The whole point of opening this issue is because I don't feel the vocabulary in Hsoub's translation reflects how Arab programmers speak Arabic or even write it (for example in apps like Slack or PR reviews); even when using Modern Standard Arabic, in their day to day work; this is coming only from my limited experience with Arab programmers from a number of Arab countries. 

I am not trying to say by any means that we should avoid or use any form of Arabic, but I felt that it was important to get the conversation started on this topic so that we as an open source community make an informed choice and get the opinion of other Arab programmers so this repository is written in a way that works for them, since they are the intended audience.
